lib/runtime-core/src/import.rs

```diff
@@ -97,6 +100,56 @@ impl ImportObject {
             map: Rc::clone(&self.map),
         }
     }
+
+    fn get_objects(&self) -> VecDeque<(String, String, Export)> {
+        let mut out = VecDeque::new();
+        for (name, ns) in self.map.borrow().iter() {
+            for (id, exp) in ns.get_exports() {
+                out.push_back((name.clone(), id, exp));
+            }
+        }
+        out
+    }
+}
+
+pub struct ImportObjectIterator {
+    elements: VecDeque<(String, String, Export)>,
+}
+
+impl Iterator for ImportObjectIterator {
+    type Item = (String, String, Export);
+    fn next(&mut self) -> Option<Self::Item> {
+        self.elements.pop_front()
+    }
+}
+
+impl IntoIterator for ImportObject {
+    type IntoIter = ImportObjectIterator;
+    type Item = (String, String, Export);
+
+    fn into_iter(self) -> Self::IntoIter {
+        ImportObjectIterator {
+            elements: self.get_objects(),
+        }
+    }
+}
+
+impl Extend<(String, String, Export)> for ImportObject {
+    fn extend<T: IntoIterator<Item = (String, String, Export)>>(&mut self, iter: T) {
+        let mut map = self.map.borrow_mut();
+        for (ns, id, exp) in iter.into_iter() {
+            if let Some(like_ns) = map.get_mut(&ns) {
+                like_ns.maybe_insert(&id, exp).expect(&format!(
```

Comment:
I don't think we should panic if the imports are overwritten. The same way extend doesn't panic on extending a `HashMap`.